tests/e2e/storage_policy_utils.go

```diff
@@ -0,0 +1,153 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/onsi/gomega"
+	"github.com/vmware/govmomi/object"
+	"github.com/vmware/govmomi/pbm"
+	"github.com/vmware/govmomi/pbm/types"
+	"github.com/vmware/govmomi/vapi/tags"
+	"k8s.io/kubernetes/test/e2e/framework"
+)
+
+func createVmfsStoragePolicy(ctx context.Context, pbmClient *pbm.Client, allocationType string,
+	categoryTagMap map[string]string) (*types.PbmProfileId, string) {
+	s1 := rand.NewSource(time.Now().UnixNano())
+	r1 := rand.New(s1)
+	profileName := fmt.Sprintf("vmfs-policy-%v-%v", time.Now().UnixNano(), strconv.Itoa(r1.Intn(1000)))
+	pbmCreateSpec := pbm.CapabilityProfileCreateSpec{
+		Name:        profileName,
+		Description: "VMFS test policy",
+		Category:    "REQUIREMENT",
+		CapabilityList: []pbm.Capability{
+			{
+				ID:        "VolumeAllocationType",
+				Namespace: "com.vmware.storage.volumeallocation",
+				PropertyList: []pbm.Property{
+					{
+						ID:       "VolumeAllocationType",
+						Value:    "Fully initialized",
+						DataType: "string",
+					},
+				},
+			},
+		},
+	}
+	for k, v := range categoryTagMap {
+
+		pbmCreateSpec.CapabilityList = append(pbmCreateSpec.CapabilityList, pbm.Capability{
+			ID:        k,
+			Namespace: "http://www.vmware.com/storage/tag",
+			PropertyList: []pbm.Property{
+				{
+					ID:       "com.vmware.storage.tag." + k + ".property",
+					Value:    v,
+					DataType: "set",
+				},
+			},
+		})
+	}
+	createSpecVMFS, err := pbm.CreateCapabilityProfileSpec(pbmCreateSpec)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+	profileID, err := pbmClient.CreateProfile(ctx, *createSpecVMFS)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+	framework.Logf("VMFS profile with id: %v and name: '%v' created", profileID.UniqueId, profileName)
+
+	return profileID, profileName
+}
+
+//deleteStoragePolicy deletes the given storage policy
+func deleteStoragePolicy(ctx context.Context, pbmClient *pbm.Client, profileID *types.PbmProfileId) {
+	_, err := pbmClient.DeleteProfile(ctx, []types.PbmProfileId{*profileID})
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+}
+
+//createCategoryNTag create one category and one tag
+func createCategoryNTag(ctx context.Context, catName string, tagName string) (string, string) {
+	tagSpec := tags.Category{Name: catName, Cardinality: "MULTIPLE"}
+	mgr := newTagMgr(ctx, e2eVSphere.Client)
+	catID, err := mgr.CreateCategory(ctx, &tagSpec)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	tagID, err := mgr.CreateTag(ctx, &tags.Tag{Name: tagName, CategoryID: catID})
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	return catID, tagID
+}
+
+//deleteCategoryNTag given category and tag
```

Comment:
Please correct the comment saying deleted given category and tag

Reply:
Done